src/test/java/org/opensrp/repository/postgres/EventsRepositoryTest.java

```diff
@@ -645,5 +648,24 @@ public void testCountEvents() {
 			eventsRepository.safeRemove(event);
 		assertEquals(0, eventsRepository.countEvents(eventSearchBean).longValue());
 	}
+
+	@Test
+	public void testFindEventsByEntityIdAndPlan() {
+		Obs obs = new Obs("concept", "decimal", "1730AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA", null, "3.5", null, "weight");
+		Event event = new Event().withBaseEntityId("4355345345431").withEventType("GrowthMonitoring").
+				withFormSubmissionId("gjhg34534nvbnv33453450").withEventDate(new DateTime()).withObs(obs)
+				.withLocationId("test-location-id").withChildLocationId("test-child-location-id");
+		event.setTeam("team");
+		event.setTeamId("team-id");
+		event.setProviderId("provider-id");
+		event.setServerVersion(12345678l);
+
+		Map<String, String> details = new HashMap<>();
+		details.put("planIdentifier", "plan-id-12345");
+		event.setDetails(details);
+		eventsRepository.add(event);
+		List<QuestionnaireResponse> questionnaireResponses = eventsRepository.findEventsByEntityIdAndPlan("4355345345431","plan-id-12345");
+		assertEquals(1,questionnaireResponses.size());
```

Comment:
assert that the correct event was returned i.e questionnaireResponses.get(0).getId() equals  event formSubmissionId

Reply:
Added assertion